app/policies/user_policy.rb

```diff
@@ -0,0 +1,29 @@
+class UserPolicy
+  attr_reader :current_user, :model
+
+  def initialize(current_user, model)
+    @current_user = current_user
+    @user = model
+  end
+
+  def index?
+    @current_user == @user
```

Comment:
Lets do that only admin can access users CRUDS.
Current User will have a separate `Profile` page.